esphome/components/i2s_audio/i2s_audio.cpp

```diff
@@ -9,6 +9,14 @@ namespace i2s_audio {
 
 static const char *const TAG = "i2s_audio";
 
+#if defined(USE_ESP_IDF) && (ESP_IDF_VERSION_MAJOR >= 5)
+#if SOC_I2S_NUM > 1
+static const uint8_t I2S_NUM_MAX = 2;  // because IDF 5+ took this away :(
+#else
+static const uint8_t I2S_NUM_MAX = 1;  // because IDF 5+ took this away :(
+#endif
```

Comment:
Wouldn't this be simpler?
```suggestion
static const uint8_t I2S_NUM_MAX = SOC_I2S_NUM;  // because IDF 5+ took this away :(
```

Reply:
It would -- I wasn't sure about the best/preferred approach here. 😄